### PR TITLE
Fisrt app version 2

### PR DESCRIPTION
**Pull request by dileepaHsenid**

### CODE REVIEW

File: first-app/src/main/java/com/hsenid/first_app/Services/StudentService.java [hunk 1]
```diff
@@ -16,36 +23,74 @@ public class StudentService {
...
+     *
+     * @param id The ID of the student to retrieve.
+     * @return The StudentResponse DTO containing student details.
+     * @throws ApiRequestException if the student with the given ID is not found.
      */
-    public Student getStudentById(String id) {
-        return studentRepo.findById(id).orElse(null);
+    public StudentResponse getStudentById(String id) {
+        // Retrieve the student from the repository by their ID
+        Optional<Student> studentOptional = studentRepo.findById(id);
```
Comment (ShehanTharaka) [inline first-app/src/main/java/com/hsenid/first_app/Services/StudentService.java]: You can use the orElseThrow method

File: first-app/src/main/java/com/hsenid/first_app/Services/StudentService.java [hunk 2]
```diff
@@ -16,36 +23,74 @@ public class StudentService {
...
     }
 
-    /**
+    /*
      * Method to remove a student by their ID
      * @param id the ID of the student to be removed
      */
     public void removeStudent(String id) {
+        // Retrieve the student from the repository by their ID
+        Optional<Student> studentOptional = studentRepo.findById(id);
```
Comment (ShehanTharaka) [inline first-app/src/main/java/com/hsenid/first_app/Services/StudentService.java]: You can use orElseThrow method

File: first-app/src/main/java/com/hsenid/first_app/dto/mapper/StudentMapper.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+     * @return the converted StudentResponse object.
+     */
+    public static StudentResponse studentToDTO(Student student) {
+        // Create a new instance of StudentResponse
+        // Set the ID and the name of the StudentResponse from the Student object
+        StudentResponse studentResponse;
+        studentResponse = new StudentResponse(student.getId(), student.getName());
+
+        // Return the populated StudentResponse object
+        return studentResponse;
```
Comment (ShehanTharaka) [inline first-app/src/main/java/com/hsenid/first_app/dto/mapper/StudentMapper.java]: Write this codes in one line

File: first-app/src/main/java/com/hsenid/first_app/dto/mapper/StudentMapper.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+     * @param studentRequest the StudentResponse DTO to be converted.
+     * @return the converted Student object.
+     */
+    public static Student dtoToStudent(StudentRequest studentRequest) {
+       Student student = new Student();
+       student.setId(studentRequest.id());
+       student.setName(studentRequest.name());
+       student.setDateOfBirth(studentRequest.dob());
+       student.setAverage(studentRequest.average());
+       return student;
```
Comment (ShehanTharaka) [inline first-app/src/main/java/com/hsenid/first_app/dto/mapper/StudentMapper.java]: Write this codes in one line

**Comment by dileepaHsenid** [comment]
Done Fixed the Problems.

**Pull request merged**